frontend/src/authentication/components/AuthGuard.js

```diff
@@ -23,6 +31,22 @@ export const AuthGuard = (props) => {
     return <Navigate to={requestedLocation} />;
   }
 
+  if (
+    sessionStorage.getItem('window-location') &&
```

Comment:
Could you please add short comments above these two if-statements to show to explain, what exactly we check this way?
Smth like 
`if the current url is not the same as the target one`